src/components/application_manager/src/commands/mobile/perform_audio_pass_thru_request.cc

```diff
@@ -365,6 +387,93 @@ bool PerformAudioPassThruRequest::IsWaitingHMIResponse() {
   return awaiting_tts_speak_response_ || awaiting_ui_response_;
 }
 
+void PerformAudioPassThruRequest::ProcessAudioPassThruIcon(
+    ApplicationSharedPtr app) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  audio_pass_thru_icon_exists_ = true;
+  if ((*message_)[strings::msg_params].keyExists(
+          strings::audio_pass_thru_icon)) {
+    if (IsAudioPassThruIconParamValid()) {
+      const std::string& icon_storage_file = GetAudioPassThruIconFilename(app);
+
```

Comment:
@dtrunov I can use this function to check if the image exists in the file system. String value parameter requires checking for special characters. Please, look at the changes.